cmake-modules/AzureDoxygen.cmake

```diff
@@ -19,7 +19,16 @@ function(generate_documentation PROJECT_NAME PROJECT_VERSION)
         set(DOXYGEN_RECURSIVE YES)
         set(DOXYGEN_USE_MDFILE_AS_MAINPAGE ./README.md)
         set(DOXYGEN_INLINE_SOURCES YES)
+        # Explicitly adding exception files to be skipped.
+        set(DOXYGEN_EXCLUDE_PATTERNS
+            json.hpp
+            test
+            Details
+            curl_connection_pool_private.hpp
```

Comment:
Is it possible to use a `*_internal` and `*_private` regex to filter what shouldn't show up as a convention, rather than hardcoding specific file names?

Reply:
Would `*_private*` work, as an `DOXYGEN_EXCLUDE_PATTERNS`?

Reply:
Would a file my_object_details.hpp be excluded?  We wouldn't want that.  We only want ::Details namespace to be excluded

Reply:
I like to be as much explicit as possible to avoid cases like what Rick mentioned.
I have updated just to have a private regex, but, other than that, I would prefer to keep it by file name.

For the ::Details namespace, I have added a different setting which works on Symbols level instead of files level.